browser/brave_screenshots/brave_screenshots_tab_feature.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_BRAVE_SCREENSHOTS_BRAVE_SCREENSHOTS_TAB_FEATURE_H_
+#define BRAVE_BROWSER_BRAVE_SCREENSHOTS_BRAVE_SCREENSHOTS_TAB_FEATURE_H_
+
+#include <memory>
+
+#include "base/memory/weak_ptr.h"
+
+namespace content {
+class WebContents;
+}  // namespace content
+
+namespace image_editor {
+struct ScreenshotCaptureResult;
+}  // namespace image_editor
+
+class Browser;
+
+namespace brave_screenshots {
+
+class BraveScreenshotStrategy;
+
+enum ScreenshotType {
+  kSelection,
+  kViewport,
+  kFullPage,
+};
+
+class BraveScreenshotsTabFeature {
+ public:
+  BraveScreenshotsTabFeature();
+  BraveScreenshotsTabFeature(const BraveScreenshotsTabFeature&) = delete;
+  BraveScreenshotsTabFeature& operator=(const BraveScreenshotsTabFeature&) =
+      delete;
+  ~BraveScreenshotsTabFeature();
+
+  void StartScreenshot(Browser* browser, ScreenshotType type);
+
+ private:
+  void OnCaptureComplete(const image_editor::ScreenshotCaptureResult& result);
+  std::unique_ptr<BraveScreenshotStrategy> CreateStrategy(ScreenshotType type);
+  base::WeakPtr<Browser> browser_ = nullptr;
```

Comment:
we shouldn't have `Browser` here as discussed, but having a weakptr to `Browser` in a tab feature is also pointless because the lifetime of `Browser` is necessarily longer than any given tab